src/tools/lib/python/particleCalorimeter.py

```diff
@@ -0,0 +1,245 @@
+"""
+This file is part of PIConGPU.
+
+PIConGPU is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+PIConGPU is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with PIConGPU.
+If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2017 Marco Garten, Axel Huebl
+Authors: Axel Huebl
+License: GPLv3+
+"""
+import numpy as np
+import matplotlib as mpl
+from matplotlib import pyplot as plt
+import h5py
+
+
+class ParticleCalorimeter:
+    """
+    A binned calorimeter of the amount of kinetic energy per solid angle and
+    energy-per-particle. The solid angle bin is solely determined by the
+    particle's momentum vector and not by its position, so we are emulating
+    a calorimeter at infinite distance. The calorimeter takes into account all
+    existing particles as well as optionally all particles which have already
+    left the global simulation volume.
+    """
+
+    def __init__(self,
+                 species_name="ph", sim="./",
+                 period=1000, num_bins_yaw=64, num_bins_pitch=64,
+                 num_bins_energy=1024, min_energy=10, max_energy=10000,
+                 logscale=False, opening_yaw=360, opening_pitch=180,
+                 pos_yaw=0, pos_pitch=0):
+        """
+        Parameters
+        ----------
+        species_name : string
+            name of the species for which the particle calorimeter
+            should be plotted, e.g. "ph"
+        sim : string
+            path to simOutput directory
+        period : unsigned integer
+            output periodicity of the histogram
+        num_bins_yaw : unsigned integer
+            number of bins resolving the yaw angle
+        num_bins_pitch : unsigned integer
+            number of bins resolving the pitch angle
+        num_bins_energy : unsigned integer
+            number of bins resolving the detected energies
+        min_energy : float
+            minimum detectable energy in keV
+        max_energy : float
+            maximum detectable energy in keV
+        logscale : boolen
+            if ``True`` the energy bins are logarithmically spaced
```

Comment:
removed it since the data won't be changed afterwards anymore, anyway
only the plotting could be done differently but the user can decide that for him-/herself